autoload/neomake/makers/typescript_project.vim

```diff
@@ -0,0 +1,13 @@
+" vim: ts=4 sw=4 et
+
+function! neomake#makers#typescript_project#typescript_project()
+    return {
+        \ 'exe': 'tsc',
+        \ 'args': ['--noEmit'],
+        \ 'errorformat':
+            \ '%E%f %#(%l\,%c): error %m,' .
+            \ '%E%f %#(%l\,%c): %m,' .
+            \ '%Eerror %m,' .
+            \ '%C%\s%\+%m'
+        \ }
+endfunction
```

Comment:
> How to use: 
`:NeomakeProject typescript_project`

The "project" makers are in neomake/makers and the "file" makers are neomake/makers/ft.  I modeled this after the cargo maker.

The error format is the same because tsc spits out the same errors regardless of if it's compiling a single file, or a project.

The reason that `-m commonjs` is missing is because the tsconfig file specifies the module type.